### ISSUE TITLE
OVH Hosted Exchange: Client Whitelist

### ISSUE THREAD

**Issue by iLLogical2007** [schweikert/postgrey#47]
When sending from a OVH Hosted Exchange account, OVH tries to resend the rejected mail with another outgoing mail server. Therefore the mail can be stuck in a "greylisting loop". To avoid problems with the OVH mail-out servers, I added the following regular expression exception to the client whitelist:

/^[0-9]{1,}.mo[0-9]{1,}.mail-out.ovh.net$/

The regex has been verified and is working.

Log lines before adding regex to client whitelist:
```
Feb 25 10:07:20 web01 postgrey[2237]: action=greylist, reason=new, client_name=8.mo64.mail-out.ovh.net, client_address=<IP>, sender=<SENDER>, recipient=<TARGET>

Feb 25 10:07:20 web01 postfix/smtpd[30157]: NOQUEUE: reject: RCPT from 8.mo64.mail-out.ovh.net[<IP>]: 450 4.2.0 <TARGET>: Recipient address rejected: Greylisted, see http://postgrey.schweikert.ch/help/target.html; from=<SENDER> to=<TARGET> proto=ESMTP helo=<8.mo64.mail-out.ovh.net>

Feb 25 10:17:15 web01 postgrey[2237]: action=greylist, reason=new, client_name=10.mo64.mail-out.ovh.net, client_address=<IP>, sender=<SENDER>, recipient=<TARGET>

Feb 25 10:17:15 web01 postfix/smtpd[30157]: NOQUEUE: reject: RCPT from 10.mo64.mail-out.ovh.net[<IP>]: 450 4.2.0 <TARGET>: Recipient address rejected: Greylisted, see http://postgrey.schweikert.ch/help/target.html; from=<SENDER> to=<TARGET> proto=ESMTP helo=<10.mo64.mail-out.ovh.net>
```

After adding the regex to client whitelist:
`Feb 25 10:37:16 web01 postgrey[1087]: action=pass, reason=client whitelist, client_name=2.mo64.mail-out.ovh.net, client_address=<IP>, sender=<SENDER>, recipient=<TARGET>
`

**Comment by iLLogical2007** [comment]
Found another pattern:
`/^mo[0-9]{1,}.mail-out.ovh.net$/`

Alternatively you could modify the regular expression above to a broader matching range.

**Comment by bushuew** [comment]
the same problem

...

20:31:04 reject: RCPT from 5.mo178.mail-out.ovh.net[46.105.51.53]:
20:38:40 reject: RCPT from 4.mo178.mail-out.ovh.net[46.105.49.171]:
20:48:42 reject: RCPT from 8.mo178.mail-out.ovh.net[46.105.74.227]:
21:08:40 reject: RCPT from 10.mo178.mail-out.ovh.net[46.105.76.150]:
21:48:40 reject: RCPT from mo178.mail-out.ovh.net[178.32.228.178]:
22:58:40 reject: RCPT from 2.mo178.mail-out.ovh.net[46.105.39.61]:

**Comment by iLLogical2007** [comment]
You should be fine with adding the following line to your client whitelist:
`/^(\d+.)?mo\d+.mail-out.ovh.net$/`

This regular expression should combine both expressions I mentioned above and matches all hostnames in your log excerpt (see http://regexr.com/3fgs3).

**Comment by schweikert** [comment]
implemented in 78483623d494a4de9cebb59d452d14141852f263
